Firmware/ultralcd.cpp

```diff
@@ -94,7 +94,7 @@ static float manual_feedrate[] = MANUAL_FEEDRATE;
 /* LCD message status */
 static LongTimer lcd_status_message_timeout;
 static uint8_t lcd_status_message_level;
-static char lcd_status_message[LCD_WIDTH + 1] = WELCOME_MSG;
+static char lcd_status_message[LCD_WIDTH + 1];
```

Comment:
I don't really understand how this saved 1B of RAM

Reply:
Perhaps a clue:
If we initialize the array with all elements as 1: it takes 1 extra byte of SRAM
`static char lcd_status_message[LCD_WIDTH + 1] = {1};`

If we initialize the array with all elements as 0: it takes no extra SRAM. 
`static char lcd_status_message[LCD_WIDTH + 1] = {0};`

Reply:
I think I know why this saved 1B of ram. It's because when you initialize the array with {1} it goes into the .data section, whch is in turn stored in flash. When you initialize the variable with {0} or don't initialize at all (in a global context), the variable gets placed in the .bss section which is initialized to 0x00 before jumping to main.
I see that the .data section is aligned to 2B while .bss is not. I think that's where a byte gets lost.

Reply:
That explanation sounds reasonable 👍